### PR TITLE
🔨 - Dynamically use Endpoints

### PR DESCRIPTION
**Pull request by burkematthew**
This PR creates general methods for getting data from an endpoint using settings configured in a constant dictionary.  The settings include the following settings:

* apiVersion
* endpoint
* beginningDate
* endingDate
* includeCreatedDate

`beginningDate` is used to specify a potential beginning date query parameter in the API request.  If there is none, we set this to `None`.
`endingDate` is used to specify a potential ending date query parameter in the API request.  If there is none, we set this to `None`.
`includeCreatedDate` is used to toggle whether the metadata object of the response includes `createdDate`.

### CODE REVIEW

File: mintapi/api.py [hunk 1]
```diff
@@ -520,6 +516,29 @@ def _flatten_utilization(self, data):
                 )
         return utilization
 
+    def __find_endpoint(self, name):
```
Comment (mrooney) [inline mintapi/api.py:519]: It seems like the endpoints is an array and each one has a "key", and then we have to find it inside. Is there a reason to not make ENDPOINTS a dict with a key so it actually is a key and you can just grab the rest of the values from it?

Comment (burkematthew) [reply mintapi/api.py:519]: @mrooney very good point! I can do that.

Comment (burkematthew) [reply mintapi/api.py:424]: @mrooney done!

**Review by mrooney — APPROVED** [review]
Looks great, thanks!

**Pull request merged**